lib/generators/administrate/install/install_generator.rb

```diff
@@ -10,6 +16,18 @@ class InstallGenerator < Rails::Generators::Base
 
       class_option :namespace, type: :string, default: "admin"
 
+      def model_check
+        if database_models.none?
+          puts "ERROR: Add models before installing Administrate."
+        else
+          run_routes_generator
+          create_dashboard_controller
+          run_dashboard_generators
+        end
+      end
```

Comment:
I'm not sure about this implementation. I have these concerns:

1. A lack of models will display an error on `stdout` instead of `stderr`, and it will not result in a non-zero [exit status](https://en.wikipedia.org/wiki/Exit_status) returned to the shell.
2. Also this won't be an error internally to Ruby, which will probably cause it not to play well if ran as part of a "parent" generator.
3. It is not idiomatic of Rails generators, bundling all steps into a single method.

Looking into how this should be done in the realm of Thor scripts or Rails generators... I don't see much information! I had to go check Rails's source code, and I've seen a couple of examples that raise [`Rails::Generators::Error`](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/base.rb#L14), like this one:

https://github.com/rails/rails/blob/e44b3419d44b5fa6e66305fe703f34bb63932593/railties/lib/rails/generators/actions/create_migration.rb#L62-L65

It might even be worth adding a `say_status` line like in that example, where it generates an output line like this one (but with colours):

```
    conflict    db/migrate/20201008102310_create_things.rb
Another migration is already named create_things: /home/pablobm/Downloads/deletemes/rails-deleteme/db/migrate/20201008102256_create_things.rb. Use --force to replace this migration or --skip to ignore conflicted file.
```

In practical terms, what I suggest means:

1. Make `model_check` a method that just checks if there are models and does `raise Error, message` if not.
2. Leave the other methods where they were: public and not called directly. If I recall correction, the generator will call them in order if there aren't previous errors.

Reply:
interesting, I didn't know how a raised error would work here, I'll give it a go tomorrow or friday

Reply:
Looks good now!